integration-tests/src/nft_marketplace.rs

```diff
@@ -135,3 +137,105 @@ fn create_class_should_fail_when_relay_chain_location_not_registered() {
 		);
 	});
 }
+
+#[test]
+fn nft_permissions_should_allow_marketplace_class_creation() {
+	assert!(<basilisk_runtime::Runtime as pallet_nft::Config>::Permissions::can_create(&ClassType::Marketplace));
+	assert!(!<basilisk_runtime::Runtime as pallet_nft::Config>::Permissions::can_create(&ClassType::LiquidityMining));
+	assert!(!<basilisk_runtime::Runtime as pallet_nft::Config>::Permissions::can_create(&ClassType::Redeemable));
+	assert!(!<basilisk_runtime::Runtime as pallet_nft::Config>::Permissions::can_create(&ClassType::Auction));
+	assert!(!<basilisk_runtime::Runtime as pallet_nft::Config>::Permissions::can_create(&ClassType::HydraHeads));
+}
+
+#[test]
+fn nft_permissions_should_allow_marketplace_class_minting() {
+	assert!(<basilisk_runtime::Runtime as pallet_nft::Config>::Permissions::can_mint(&ClassType::Marketplace));
```

Comment:
Due to the generic parameters, all the function calls in this and the similar tests are a bit noisy.

How about extracting them to separate function like for example for `can_mint`:
```
fn can_mint(class_type: &ClassType) -> bool {
	<basilisk_runtime::Runtime as pallet_nft::Config>::Permissions::can_mint(class_type)
}
```

So at the end you would have something like:
```
#[test]
fn nft_permissions_should_allow_marketplace_class_minting() {
	assert!(can_mint(&ClassType::Marketplace));
	assert!(!can_mint(&ClassType::LiquidityMining));
	assert!(!can_mint(&ClassType::Redeemable));
	assert!(!can_mint(&ClassType::Auction));
	assert!(!can_mint(&ClassType::HydraHeads));
}
```

I believe that it would make the assertions more readable. And also, if something changes in the generic names, etc, then we would only need to change it in only one place.

Reply:
You are right. I will update the tests.

Reply:
Hmm. I think this change will not be necessary because after applying `test-case` crate as proposed above in your first comment, the issue will be eliminated because there will be just one assert for all enum variants.

Reply:
yes, that's a good point, we could also use the test-case crate to simplify these tests. The only thing is that the tests would be need to split up into two ones, one for negative and one positive case.

Or, we could add a boolean parameter as a test case param, and also introducing two new constants, to make it even more readable:
```
const ALLOW: bool = true;
const NOT_ALLOW: bool = false;

#[test_case(&ClassType::Marketplace, ALLOW)]
#[test_case(&ClassType::LiquidityMining, NOT_ALLOW)]
#[test_case(&ClassType::Redeemable, NOT_ALLOW)]
#[test_case(&ClassType::Auction, NOT_ALLOW)]
#[test_case(&ClassType::HydraHeads, NOT_ALLOW)]
fn nft_permissions_should_allow_marketplace_class_creation(class_type: &ClassType, is_allowed_by_nft_permission: bool) {
	assert_eq!(is_allowed_by_nft_permission, <basilisk_runtime::Runtime as pallet_nft::Config>::Permissions::can_create(class_type));
}
```

As you wish ;)